spacelift/resource_stack_activator.go

```diff
@@ -0,0 +1,129 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func resourceStackActivator() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_stack_activator` is used to to enable/disable Spacelift Stack.",
+		CreateContext: resourceStackActivatorCreate,
+		ReadContext:   resourceStackActivatorRead,
+		UpdateContext: resourceStackActivatorUpdate,
+		DeleteContext: schema.NoopContext,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:             schema.TypeString,
+				Description:      "ID of the stack to enable/disable",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Enable/disable stack",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceStackActivatorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceStackActivatorUpdate(ctx, d, meta)
+}
+
+func resourceStackActivatorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	stack, err := queryStack(ctx, d, meta)
+	if err != nil {
+		return diag.Errorf("failed to retrieve stack: %v", internal.FromSpaceliftError(err))
+	}
+	if stack == nil {
+		d.SetId("")
+		d.Set("enabled", false)
+	} else {
+		d.SetId(fmt.Sprintf("activator-%d", time.Now().Unix()))
+		d.Set("enabled", !stack.IsDisabled)
+	}
+	return nil
+}
+
+func resourceStackActivatorUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	stack, err := queryStack(ctx, d, meta)
+	if err != nil {
+		return diag.Errorf("failed to retrieve stack: %v", internal.FromSpaceliftError(err))
+	}
+	if stack == nil {
+		d.SetId("")
+	} else {
+		d.SetId(fmt.Sprintf("activator-%d", time.Now().Unix()))
+	}
+	enabled, ok := d.Get("enabled").(bool)
+	if !ok {
+		return diag.Errorf("invalid enabled attribute: %v", d.Get("enabled"))
+	}
+	if !stack.IsDisabled == enabled {
+		return nil
+	}
+	if stack.IsDisabled {
```

Comment:
Personally I think it would make more sense if you used `enabled` here - that way we're basing the action on the Terraform property value, not on the current state of the stack. I know the outcome is identical either way, it just feels slightly wrong to me as-is.